pkg/translator/loki/convert.go

```diff
@@ -42,10 +42,15 @@ var defaultExporterLabels = model.LabelSet{"exporter": "OTLP"}
 func convertAttributesAndMerge(logAttrs pcommon.Map, resAttrs pcommon.Map) model.LabelSet {
 	out := defaultExporterLabels
 
+	if resourcesToLabel, found := resAttrs.Get(hintResources); found {
+		labels := convertAttributesToLabels(resAttrs, resourcesToLabel)
+		out = out.Merge(labels)
+	}
+
 	// get the hint from the log attributes, not from the resource
 	// the value can be a single resource name to use as label
 	// or a slice of string values
-	if resourcesToLabel, found := resAttrs.Get(hintResources); found {
+	if resourcesToLabel, found := logAttrs.Get(hintResources); found {
```

Comment:
This `resourcesToLabel` overrides the one defined on line 45. You'll need different names for them and a test for this condition.

Reply:
am I miss something? `resourcesToLabel` scope only every if?

Reply:
Sorry, you are absolutely right!